lib/iex/lib/iex/autocomplete.ex

```diff
@@ -563,4 +570,95 @@ defmodule IEx.Autocomplete do
   defp extract_from_ast(_ast_node, _acc) do
     :error
   end
+
+  defp path_fragment(expr), do: collect_path_fragment(expr, [])
+
+  # collects the fragment and store it on `acc`, calls `return_path_fragment` 
+  # when it finds a `"`, discards `acc` otherwise
+  defp collect_path_fragment([], _acc), do: []
+  defp collect_path_fragment([?", ?\\ | t], acc), do: collect_path_fragment(t, [?\\, ?" | acc])
+  defp collect_path_fragment([?{, ?# | _rest], _acc), do: []
+  defp collect_path_fragment([?" | t], acc), do: return_path_fragment(t, acc)
+  defp collect_path_fragment([h | t], acc), do: collect_path_fragment(t, [h | acc])
+
+  # scans the rest of the `expr`, returns `acc` if no other `"` is found, 
+  # fallback to `discard_path_fragment` otherwise
+  defp return_path_fragment([], acc), do: acc
+  defp return_path_fragment([?", ?\\ | t], acc), do: return_path_fragment(t, [?\\, ?" | acc])
+  defp return_path_fragment([?" | t], acc), do: discard_path_fragment(t, acc)
+  defp return_path_fragment([_ | t], acc), do: return_path_fragment(t, acc)
+
+  # scans the rest of the `expr`, discards `acc` if no other `"` is found, 
+  # fallback to `return_path_fragment` otherwise
+  defp discard_path_fragment([], _acc), do: []
+  defp discard_path_fragment([?", ?\\ | t], acc), do: discard_path_fragment(t, [?\\, ?" | acc])
+  defp discard_path_fragment([?" | t], acc), do: return_path_fragment(t, acc)
+  defp discard_path_fragment([_ | t], acc), do: discard_path_fragment(t, acc)
+
+  defp expand_path(code) do
+    case path_fragment(code) do
+      [] ->
+        no()
+
+      path_fragment ->
+        path_fragment = List.to_string(path_fragment)
+        possible_paths = find_possible_paths(path_fragment)
+        expand_path(path_fragment, possible_paths)
+    end
+  end
+
+  defp expand_path(path_fragment, possible_paths) do
+    expansions =
+      Enum.map(possible_paths, fn {path, dir?} ->
+        {completion_part(path_fragment, path, dir?), String.to_charlist(Path.basename(path))}
+      end)
+
+    case expansions do
+      [] ->
+        no()
+
+      [{unique, _}] ->
+        {:yes, String.to_charlist(unique), []}
+
+      list ->
+        {completions, filenames} = Enum.unzip(list)
+        common_prefix_size = :binary.longest_common_prefix(completions)
+        hint = completions |> hd() |> binary_part(0, common_prefix_size) |> String.to_charlist()
+        {:yes, hint, filenames}
```

Comment:
We already have code that does this, I believe it is called format_expansions or similar. We should just reuse that instead.